query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/fields/connection/LdapConnectionField.java

```diff
@@ -95,4 +100,50 @@ public LdapConnectionField encryptionMethod(String encryptionMethod) {
     this.encryptionMethod.setValue(encryptionMethod);
     return this;
   }
+
+  public String getLdapUrl() {
+    String url = encryptionMethod().equalsIgnoreCase(LDAPS) ? "ldaps://" : "ldap://";
+    return String.format("%s%s:%d", url, hostname(), port());
+  }
+
+  public static class ListImpl extends BaseListField<LdapConnectionField> {
+
+    public static final String DEFAULT_FIELD_NAME = "connection";
+
+    private Callable<LdapConnectionField> newConnection;
+
+    public ListImpl() {
+      super(DEFAULT_FIELD_NAME);
+      newConnection = LdapConnectionField::new;
+    }
+
+    public ListImpl(String fieldName) {
+      super(fieldName);
+      newConnection = LdapConnectionField::new;
+    }
+
+    @Override
+    public Callable<LdapConnectionField> getCreateListEntryCallable() {
+      return newConnection;
+    }
+
+    @Override
+    public ListImpl useDefaultRequired() {
+      newConnection =
+          () -> {
+            LdapConnectionField entry = new LdapConnectionField();
+            entry.useDefaultRequired();
+            return entry;
+          };
+
```

Comment:
✏️ I think we should add `isRequired(true)` here. Can't think of a time where someone would pass in a list of connections and be okay with it being empty or null by default.